File: 백준/1번문제/최원준/Q1620.java
// https://www.acmicpc.net/problem/
package 백준;

/*


1. 아이디어 :
    포켓몬 : 번호, 번호 : 포켓몬을 추적하는 두개의 해시맵을 사용합니다.
2. 시간복잡도 :
    O( n+m )
3. 자료구조 :
    해시맵

 */
import javax.xml.stream.events.Characters;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;

class Q1620 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static String input;
    static String[] inputs;
    static int[] ninputs;

    public static void main(String[] args) throws IOException {
        HashMap<Integer, String> numToName = new HashMap<>();
        HashMap<String, Integer> nameToNum = new HashMap<>();
        inputs = br.readLine().split(" ");
        int n = Integer.parseInt(inputs[0]), m = Integer.parseInt(inputs[1]);
        for (int i = 0; i<n; i++) {
            input = br.readLine();
            numToName.put(i+1, input);
            nameToNum.put(input, i+1);
        }

        for (int i = 0; i<m; i++) {
            input = br.readLine();
            if (Character.isAlphabetic(input.charAt(0))) {
                System.out.println(nameToNum.get(input));
            } else {
                System.out.println(numToName.get(Integer.parseInt(input)));
            }
        }
    }
}


/*
import sys
#sys.setrecursionlimit(1000000)
# input = sys.stdin.readline
input = lambda: sys.stdin.readline().rstrip()

def solution(n, m):
    name_to_num = {}
    num_to_name = {}

    for i in range(n):
        pokemon = input()
        name_to_num[pokemon] = i+1
        num_to_name[i+1] = pokemon

    for i in range(m):
        query = input()
        if query[0].isnumeric():
            print(num_to_name[int(query)])
        else:
            print(name_to_num[query])

if __name__ == '__main__':
    n, m = map(int, input().split())
    solution(n, m)
 */

File: 백준/1번문제/이채현/Q1620.py
# 풀이 방법 : 딕셔너리로 접근
# dict = {1:피카츄,
#         2:야도라,
#         3:파이리}
# 뭐 이런식으로 dictionary로 만든 후,
# M개, question을 넣었을 때 들어오는 값이 숫자이면 dict[key값] 한 다음에 item인 포켓몬 이름 뱉게 코딩
# 반대로, 들어오는 값이 문자이면 dict[item값] 한 다음에 key값인 숫자를 뱉게 코딩
# 처음에는 위와 같이 했는데 비효율적인지, 시간 문제인지 계~속 오답.. 
# 그래서 찾아보니까 이렇게 하지 말고 아예 딕셔너리를 만들 때,
# dict = {1:피카츄,
#         2:야도라,
#         3:파이리,
#         피카츄:1,
#         야도라:2}
#         파이리:3}
# 이런식으로 다 만들어 버리고 숫자면 item을 피카츄를 뽑고, key값이 문자면 item을 숫자를 뽑는 식으로 코딩함
# 이렇게 하니까 해결됨!


import sys

input = sys.stdin.readline

N, M = map(int, input().split())

my_dict = {}

for i in range(1, N+1):
    a = input().rstrip()
    my_dict[i] = a
    my_dict[a] = i

for i in range(M):
    question = input().rstrip()
    if question.isdigit():
        print(my_dict[int(question)])
    else:
        print(my_dict[question])
